tests/test-vip-plugins.php

```diff
@@ -0,0 +1,164 @@
+
+		/**
+		 * Check that updating the option is OK, add an extra plugin
+		 */
+		$plugin_change = array_merge( $this->option_active_plugins, array( 'amp-wp/amp.php' ) );
+		$option_update = update_option( 'active_plugins', $plugin_change );
+		$this->assertTrue( $option_update );
+
+		/**
+		 * Check that the option still makes sense again
+		 */
+		// emulates update
+		$track_plugin_change = array_diff( $plugin_change, wpcom_vip_get_filtered_loaded_plugins() );
+		// emulates get
+		$saved_plugins = array_merge( wpcom_vip_get_filtered_loaded_plugins(), $track_plugin_change );
+		$this->assertEquals( $saved_plugins, get_option( 'active_plugins' ) );
+	}
+
+	public function test__modify_network_active_plugins() {
+		if ( ! is_multisite() ) {
+			$this->markTestSkipped( 'Not relevant on single-site' );
+			return;
+		}
+
+		/**
+		 * Clear global of code activated plugins from previous test
+		 */
+		global $vip_loaded_plugins;
+		$vip_loaded_plugins = array();
+
+		/**
+		 * Set active_sitewide_plugins to empty to start the test
+		 */
+		update_site_option( 'active_sitewide_plugins', [] );
+
+		/**
+		 * Ensure the values are indeed empty
+		 */
+		$this->assertEmpty( get_site_option( 'active_sitewide_plugins' ) );
+
+		/**
+		 * Update the option with our list of active network plugins
+		 */
+		update_site_option( 'active_sitewide_plugins' , $this->option_active_sitewide_plugins );
+
+		/**
+		 * Check that option is not empty
+		 */
+		$this->assertNotEmpty( get_site_option( 'active_sitewide_plugins' ) );
+
+		/**
+		 * Setup the code activated plugins
+		 */
+		foreach ( $this->code_activated_plugins as $plugin ) {
+			wpcom_vip_add_loaded_plugin( $plugin );
+		}
+
+		/**
+		 * Check that list of code activated plugins matches the mocked data
+		 */
+		$this->assertEquals( $this->code_activated_plugins, wpcom_vip_get_loaded_plugins() );
+
+		/**
+		 * Check that the returned option matches a merge of the filtered loaded plugins and active plugins
+		 */
+		$merged_plugins = array_merge( wpcom_vip_get_network_filtered_loaded_plugins(), $this->option_active_sitewide_plugins );
+		$this->assertEquals( $merged_plugins, get_site_option( 'active_sitewide_plugins' ) );
+
+		/**
+		 * Check that updating the option is OK, add an extra plugin
+		 */
+		$plugin_change = array_merge( $this->option_active_sitewide_plugins, array( 'amp-wp/amp.php' => 1507904134 ) );
+		$option_update = update_site_option( 'active_sitewide_plugins', $plugin_change );
+		$this->assertTrue( $option_update );
+
+		/**
+		 * Check that the option still makes sense again
+		 */
+		// emulates update
+		$track_plugin_change = array_diff( $plugin_change, wpcom_vip_get_network_filtered_loaded_plugins() );
+		// emulates get
+		$saved_plugins = array_merge( wpcom_vip_get_network_filtered_loaded_plugins(), $track_plugin_change );
+		$this->assertEquals( $saved_plugins, get_site_option( 'active_sitewide_plugins' ) );
```

Comment:
Any time we have multiple `assert*` calls in a single method, we should pass in the `message` parameter (with a unique string) to help us pinpoint where our test failed. Otherwise when this test fails, all we'll see is `array( 4, 5, 6 ) did not match array( 1, 2, 3 )` which doesn't really tell us what's wrong.

So, we can do something like this:

```
$this->assertEquals( $saved_plugins, get_site_option( 'active_sitewide_plugins' ), 'The value of the `active_sitewide_plugins` option did not match our expected `$saved_plugins`' );
```

This way, we can pinpoint that line 162 is where things failed.